src/main/java/com/epam/aidial/cfg/domain/service/ModelService.java

```diff
@@ -118,6 +137,17 @@ private void assertNewModelDisplayNameAndDisplayVersion(ModelEntity entity, Mode
         }
     }
 
+    private void assertNotConcurrencyOverwrite(ModelEntity entity, String oldHash) {
+        if (oldHash == null) {
+            return;
+        }
+        var newHash = calculator.calculateHash(mapper.toDomain(entity));
```

Comment:
it's non obvious name. i expect newHash for new model (with updated props). it's better to call it currentHash imho